metals/src/main/scala/scala/meta/internal/metals/callHierarchy/CallHierarchyProvider.scala

```diff
@@ -0,0 +1,619 @@
+                .collect { case ResolvedSymbolOccurrence(_, Some(occurence)) =>
+                  FindIncomingCallsResult(
+                    occurence,
+                    parentRange.get,
+                    List(name.pos.toLSP),
+                  )
+                }
+            case _ =>
+              name.children.flatMap { child =>
+                search(child, parent, parentRange)
+              }
+          }
+        case v: Defn.Val =>
+          v.pats match {
+            case pat :: _ => searchVal(v.pos.toLSP, pat, v.rhs)
+            case Nil => default
+          }
+        case v: Defn.Var =>
+          v.pats match {
+            case pat :: _ =>
+              v.rhs
+                .map(rhs => searchVal(v.pos.toLSP, pat, rhs))
+                .getOrElse(default)
+            case Nil => default
+          }
+        case _ => {
+          extractNameFromMember(tree) match {
+            case Some((member, name)) =>
+              tree.children.flatMap(child =>
+                search(child, Some(name), Some(member.pos.toLSP))
+              )
+            case None =>
+              default
+          }
+        }
+      }
+    }
+    FindIncomingCallsResult.group(search(root, None, None))
+  }
+
+  private def findIncomingCallsSynthetics(
+      source: AbsolutePath,
+      doc: TextDocument,
+      info: CallHierarchyItemInfo,
+  ): List[FindIncomingCallsResult] =
+    FindIncomingCallsResult.group(
+      doc.synthetics
+        .flatMap(syn =>
+          extractSelectTree(syn.tree).collect {
+            case s.SelectTree(s.OriginalTree(Some(range)), id)
+                if id.exists(id => info.symbols.contains(id.symbol)) => (
+              for {
+                tree <- trees.findLastEnclosingAt(source, range.toLSP.getStart)
+                (definition, name) <- getSpecifiedOrFindDefinition(Some(tree))
+                occurence <- this.definition
+                  .positionOccurrence(source, name.pos.toLSP.getStart, doc)
+                  .occurrence
+                  .filter(_.role.isDefinition)
+              } yield FindIncomingCallsResult(
+                occurence,
+                name.pos.toLSP,
+                List(range.toLSP),
+              )
+            )
+          }
+        )
+        .flatten
+        .toList
+    )
+
+  /**
+   * Resolve incoming calls for a given call hierarchy item.
+   */
+  def incomingCalls(
+      params: CallHierarchyIncomingCallsParams,
+      token: CancelToken,
+  )(implicit ec: ExecutionContext): Future[List[CallHierarchyIncomingCall]] = {
+    val info = params.getItem.getData
+      .asInstanceOf[JsonElement]
+      .as[CallHierarchyItemInfo]
+      .get
+
+    for {
+      paths <-
+        if (!info.isLocal)
+          references
+            .pathsMightContainSymbol(
+              params.getItem.getUri.toAbsolutePath,
+              info.symbols.toSet,
+            )
+        else Future.successful(Set.empty)
```

Comment:
```suggestion
        else Future.successful(Set.empty[AbsolutePath])
```
Nit but just so the type of `paths` isn't funky.